```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/7fwlv5d9w73q38j2x8lyf50010fyycng-dosage-2.17.tar.gz
source root is dosage-2.17
setting SOURCE_DATE_EPOCH to timestamp 1580675094 of file dosage-2.17/setup.cfg
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing setuptoolsBuildPhase
[33mWARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.[0m
[33mWARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff6cf07c0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/[0m
[33mWARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff6ccb9a0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/[0m
[33mWARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff6ccbe80>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/[0m
[33mWARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff6ccbcd0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/[0m
[33mWARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff6ccbb20>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/setuptools-scm/[0m
[31mERROR: Could not find a version that satisfies the requirement setuptools_scm (from versions: none)[0m
[31mERROR: No matching distribution found for setuptools_scm[0m
Traceback (most recent call last):
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/installer.py", line 75, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/nix/store/81lwy2hfqj4c1943b1x8a0qsivjhdhw9-python3-3.9.6/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/81lwy2hfqj4c1943b1x8a0qsivjhdhw9-python3-3.9.6/bin/python3.9', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmp6sju4uvq', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/build/dosage-2.17/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 11, in <module>
    setup(
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/__init__.py", line 152, in setup
    _install_setup_requires(attrs)
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/dist.py", line 785, in fetch_build_eggs
    resolved_dists = pkg_resources.working_set.resolve(
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/pkg_resources/__init__.py", line 766, in resolve
    dist = best[req.key] = env.best_match(
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1051, in best_match
    return self.obtain(req, installer)
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1063, in obtain
    return installer(requirement)
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/dist.py", line 844, in fetch_build_egg
    return fetch_build_egg(self, req)
  File "/nix/store/z0v20ncmp4brnas2g2a2bdprdl3im363-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/installer.py", line 77, in fetch_build_egg
    raise DistutilsError(str(e)) from e
distutils.errors.DistutilsError: Command '['/nix/store/81lwy2hfqj4c1943b1x8a0qsivjhdhw9-python3-3.9.6/bin/python3.9', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmp6sju4uvq', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.
```

--- a/pkgs/applications/graphics/dosage/default.nix
+++ b/pkgs/applications/graphics/dosage/default.nix
@@ -13,6 +13,8 @@ python3Packages.buildPythonApplication rec {
     pytestCheckHook pytest-xdist responses
   ];
 
+  nativeBuildInputs = with python3Packages; [ setuptools-scm ];
+
   propagatedBuildInputs = with python3Packages; [
     colorama imagesize lxml requests setuptools
   ];

